mm2src/coins/utxo_signer/src/with_trezor.rs

```diff
@@ -173,6 +175,16 @@ impl<TxP: TxProvider + Send + Sync> TrezorTxSigner<TxP> {
         }
     }
 
+    /// `expiry` must be set for Decred and Zcash coins *only*.
+    /// This fixes : https://github.com/KomodoPlatform/atomicDEX-API/issues/1626
+    fn expiry_if_required(&self, tx_expiry: u32) -> Option<u32> {
+        if self.is_overwinter_compatible() {
```

Comment:
The comment is about `is_overwinter_compatible`: it checks only `self.params.unsigned_tx.version > 3`, which might be insufficient. Theoretically, there can be coins that are not overwinter compatible but still use tx version >3. [Trezor JS code from comment](https://github.com/trezor/trezor-utxo-lib/blob/trezor/src/transaction.js#L405) uses additional `isZcashType` check. Can you do something similar in our code, please?

Reply:
I believe, this is equivalent to the JS implementation:
https://github.com/KomodoPlatform/atomicDEX-API/blob/0dd8425351d19def36fa5ed819f8ba2b93d2a051/mm2src/coins/utxo_signer/src/with_trezor.rs#L202